clang/lib/Format/UnwrappedLineParser.cpp

```diff
@@ -1157,7 +1157,15 @@ void UnwrappedLineParser::parsePPDefine() {
   // guard processing above, and changes preprocessing nesting.
   FormatTok->Tok.setKind(tok::identifier);
   FormatTok->Tok.setIdentifierInfo(Keywords.kw_internal_ident_after_define);
-  nextToken();
+
+  if (Style.SkipMacroDefinitionBody) {
+    do {
+      nextToken();
+    } while (!eof());
+  } else {
+    nextToken();
+  }
```

Comment:
Instead, we can simply mark the tokens `Finalized` (after `Line->InMacroBody` is set to `true` below) so that the continuation indenter can be left alone.

Reply:
Hi!
Maybe it's a little late, but I wanted to let you know that I've tried that but it didn't give the expected results. Didn't have time to dig into this approach.

Reply:
Np!